apps/projects/views.py

```diff
@@ -291,9 +292,55 @@ def is_project_view(self):
     def raise_exception(self):
         return self.request.user.is_authenticated
 
+    def get_context_data(self, **kwargs):
+        """Append insights to the template context."""
+        context = super().get_context_data(**kwargs)
+
+        modules = self.project.modules.all()
+        ideas = Idea.objects.filter(module__in=modules)
+        map_ideas = MapIdea.objects.filter(module__in=modules)
+        proposals = Proposal.objects.filter(module__in=modules)
+        comments = get_all_comments_project(project=self.project)
+
+        ratings_ideas = Rating.objects.filter(idea__in=ideas)
+        ratings_map_ideas = Rating.objects.filter(mapidea__in=map_ideas)
+        ratings_comments = Rating.objects.filter(comment__in=comments)
```

Comment:
@hklarner here you are hitting the DB again, that is why I had the prefetch in the comments with the ratings, and also now you are including the ratings that have value 0. Why would you remove the `Ratings.objects.exclude(0)`?
ref https://docs.djangoproject.com/en/3.2/ref/models/querysets/#prefetch-related

Reply:
btw the `.filter()` is lazy, it doesn't go against the DB, see
- https://docs.djangoproject.com/en/4.2/topics/db/queries/#querysets-are-lazy